app/controllers/api/groups_controller.rb

```diff
@@ -161,6 +161,86 @@ def update_marks
         HttpStatusHelper::ERROR_CODE['message']['200'] }, status: 200
     end
 
+    def add_extra_marks
+      assignment = Assignment.find(params[:assignment_id])
+      if assignment.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+            'No assignment exists with that id' }, status: 404
+        return
+      end
+
+      group = Group.find(params[:id])
+      if group.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+            'No group exists with that id' }, status: 404
+        return
+      end
+      if group.grouping_for_assignment(params[:assignment_id])
+              .has_submission?
+        result = group.grouping_for_assignment(params[:assignment_id])
+                      .current_submission_used
+                      .get_latest_result
+      else
+        render 'shared/http_status', locals: { code: '404', message:
+            'No submissions exist for that group' }, status: 404
+        return
+      end
+      extra_mark = ExtraMark.new(result_id: result.id, extra_mark: params[:extra_marks],
+                                 description: params[:description], unit: ExtraMark::POINTS)
+      unless extra_mark.save
+        # Some error occurred
+        render 'shared/http_status', locals: { code: '500', message:
+        extra_mark.errors.full_messages.first }, status: 500
+        return
+      end
+      extra_mark.save
+      render 'shared/http_status', locals: { code: '200', message:
+          HttpStatusHelper::ERROR_CODE['message']['200'] }, status: 200
+    end
+
+    def remove_extra_marks
+      assignment = Assignment.find(params[:assignment_id])
+      if assignment.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+            'No assignment exists with that id' }, status: 404
+        return
+      end
+
+      group = Group.find(params[:id])
+      if group.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+            'No group exists with that id' }, status: 404
+        return
+      end
+      if group.grouping_for_assignment(params[:assignment_id])
+              .has_submission?
+        result = group.grouping_for_assignment(params[:assignment_id])
+                      .current_submission_used
+                      .get_latest_result
+      else
+        render 'shared/http_status', locals: { code: '404', message:
+            'No submissions exist for that group' }, status: 404
+        return
+      end
+      extra_mark = ExtraMark.find_by(result_id: result.id,
+                                     description: params[:description],
+                                     extra_mark: params[:extra_marks])
```

Comment:
Don't you think it's overkill to have to specify three attributes in order to find a single extra mark?

Reply:
1. A result can have more than 1 extra mark 
2. If 2 records have same result id and same extra mark but different description
3. If 2 records have same result id and same description but different extra mark

example:
result_id: 20, description: bonus, extra_mark: 10
result_id: 20, description: bonus for good score, extra_mark: 10
result_id: 20, description: bonus, extra_mark: 2

I think we need to specify these three fields to find a single extra mark. If there is any other way or any other specific reason not to include, I am ready to work on that too.

Reply:
Ok good point. In general we should be specifying extra_marks based on the `id`. Getting that data is going to require another API call though. Let's leave this for now and I'll think about how to reconfigure things after this PR gets pulled in.